internal/controller/multiclusterservice_controller.go

```diff
@@ -192,9 +321,43 @@ func (r *MultiClusterServiceReconciler) reconcileDelete(ctx context.Context, mcs
 	return ctrl.Result{}, nil
 }
 
+// requeueSveltosProfileForClusterSummary asserts that the requested object has Sveltos ClusterSummary
+// type and requeues its owner, which is either a Sveltos Profile or ClusterProfile object.
+func requeueSveltosProfileForClusterSummary(ctx context.Context, o client.Object) []reconcile.Request {
+	l := ctrl.LoggerFrom(ctx)
+	msg := "cannot queue request"
+
+	cs, ok := o.(*sveltosv1beta1.ClusterSummary)
+	if !ok {
+		l.Error(fmt.Errorf("request is not for a ClusterSummary object"), msg, "Name", o.GetName(), "Namespace", o.GetNamespace())
+		return []ctrl.Request{}
+	}
+
+	ownerRef, err := sveltosv1beta1.GetProfileOwnerReference(cs)
+	if err != nil {
+		l.Error(err, msg, "Name", o.GetName(), "Namespace", o.GetNamespace())
+		return []ctrl.Request{}
+	}
+
+	// The Profile/ClusterProfile object has the same name as its
+	// owner object which is either ManagedCluster or MultiClusterService.
+	req := client.ObjectKey{Name: ownerRef.Name}
+	if ownerRef.Kind == sveltosv1beta1.ProfileKind {
+		req.Namespace = o.GetNamespace()
+	}
+
+	return []ctrl.Request{{NamespacedName: req}}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *MultiClusterServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&hmc.MultiClusterService{}).
+		For(&hmc.MultiClusterService{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&sveltosv1beta1.ClusterSummary{},
+			handler.EnqueueRequestsFromMapFunc(requeueSveltosProfileForClusterSummary),
+			builder.WithPredicates(predicate.Funcs{
```

Comment:
the same questions as of `managedcluster-ctrl`: do we need the `generic` event here and do we need to trigger the reconciliation if a `clustersummary` object gets its status update?

Reply:
Same answer.